aiidalab_launch/profile.py

```diff
@@ -70,9 +71,44 @@ def __post_init__(self):
         if self.home_mount is None:
             self.home_mount = f"{CONTAINER_PREFIX}{self.name}_home"
 
+        # Normalize extra mount mode to be "rw" by default
+        # so that we match Docker default but are explicit.
+        for i, extra_mount in enumerate(self.extra_mounts):
+            if len(extra_mount.split(":")) == 2:
+                self.extra_mounts[i] = f"{extra_mount}:rw"
+            self.parse_extra_mount(self.extra_mounts[i])
+
     def container_name(self) -> str:
         return f"{CONTAINER_PREFIX}{self.name}"
 
+    def parse_extra_mount(
+        self, extra_mount: str
+    ) -> tuple[Path, PurePosixPath, str | None]:
+        fields = extra_mount.split(":")
+        if len(fields) < 2 or len(fields) > 3:
+            raise ValueError(f"Invalid extra mount option '{extra_mount}'")
+
+        source, target = fields[:2]
+        source_path, target_path = Path(source), PurePosixPath(target)
+        if source_path.is_absolute() and not source_path.exists():
+            raise ValueError(f"Directory '{source}' does not exist")
+
+        # We do not allow relative paths so if the path is not absolute,
+        # we assume volume mount, whose name is restricted by Docker.
+        if not source_path.is_absolute() and not re.match(
+            r"[a-zA-Z0-9][a-zA-Z0-9_.-]+$", source
```

Comment:
I couldn't find this in any Docs, but Docker actually prints this regex in an error message. :-)
This validation could be added to the home mount as well.

Reply:
Nice find! 👍

Reply:
Can we add a `valid_volume_name` function to the `util` module and then use it for the other volume names as well?

Reply:
I have decided to keep this function in profile instead of util, because almost the same regex already existed for validating the profile names so it made sense to me to have both of them at the same place.